osu.Game/Screens/OnlinePlay/Multiplayer/Match/Playlist/MultiplayerQueueList.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Rooms;
+using osuTK;
+
+namespace osu.Game.Screens.OnlinePlay.Multiplayer.Match.Playlist
+{
+    /// <summary>
+    /// A gameplay-ordered list of <see cref="DrawableRoomPlaylistItem"/>s.
+    /// </summary>
+    public class MultiplayerQueueList : DrawableRoomPlaylist
+    {
+        public MultiplayerQueueList()
+            : base(false, false, true)
+        {
+        }
+
+        protected override FillFlowContainer<RearrangeableListItem<PlaylistItem>> CreateListFillFlowContainer() => new QueueFillFlowContainer
+        {
+            Spacing = new Vector2(0, 2)
+        };
+
+        private class QueueFillFlowContainer : FillFlowContainer<RearrangeableListItem<PlaylistItem>>
+        {
+            [Resolved(typeof(Room), nameof(Room.QueueMode))]
+            private Bindable<QueueMode> queueMode { get; set; }
+
+            [Resolved(typeof(Room), nameof(Room.Playlist))]
+            private BindableList<PlaylistItem> roomPlaylist { get; set; }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+                queueMode.BindValueChanged(_ => InvalidateLayout());
+            }
+
+            public override IEnumerable<Drawable> FlowingChildren
+            {
+                get
+                {
+                    switch (queueMode.Value)
+                    {
+                        default:
+                            return AliveInternalChildren.Where(d => d.IsPresent)
```

Comment:
Is there a reason these aren't using `base.FlowingChildren` instead, to avoid re-specifying the base conditionals here? Just to avoid the ordering at the base implementation (not sure how much that matters when we're re-ordering here).

Reply:
This is no longer implemented here.